controllers/snapshot/snapshot_adapter.go

```diff
@@ -88,6 +88,25 @@ func (a *Adapter) EnsureAllIntegrationTestPipelinesExist() (results.OperationRes
 	return results.ContinueProcessing()
 }
 
+// EnsureGlobalComponentImageUpdated is an operation that ensure the ContainerImage in the Global Candidate List
+// being updated when the ApplicationSnapshot passed all the integration tests
+func (a *Adapter) EnsureGlobalComponentImageUpdated() (results.OperationResult, error) {
+	if gitops.HaveHACBSTestsSucceeded(a.snapshot) {
+		patch := client.MergeFrom(a.component.DeepCopy())
+		for _, component := range a.snapshot.Spec.Components {
+			if component.Name == a.component.Name {
```

Comment:
We will also need to check if `a.component` is `nil` because not all ApplicationSnapshots are associated with a single component.

Reply:
We may do that check even earlier, at [line 94](https://github.com/redhat-appstudio/integration-service/pull/35/files#diff-8e718c6bf06f754db7e9e94c6f74b5c6704941dff95d637289e3c4cc194c95a0R94)

Reply:
> We will also need to check if `a.component` is `nil` because not all ApplicationSnapshots are associated with a single component.
Do you mean to check "a.snapshot.Spec.Components" which are associated with the snapshot?

Reply:
We just need to check the `a.component`.